### PR TITLE
[Magiclysm] Component spell content update

### PR DESCRIPTION
**Pull request by ghost**
#### Summary

SUMMARY: Mods "Spells using components system"

#### Purpose of change

Follow-up to #42265. 

#### Describe the solution

Add 7/8 new spells, and alter an existing one.

Knifeshot: Requested by KorG, accelerate a knife at high speed to deal high pierce damage. Maxxing the spell out makes it go fast enough to be bullet damage, and pierce through enemies. Knives limited to steel or iron fixed blades with a point.

Decaying Boneclub: Bone club with additional biological damage.

Flamesword: Set a wooden sword on fire

Flamebreath: Invoke a dust explosion to create a cone of fire. Longer, narrower range with a DoT compared to Burning Hands.

Feral Form: Wrap bones in fur and shatter it, becoming beastly with +STR/DEX, -INT/PER, Claws, Fur, and Adrenaline.

Impact Sling: Infuse a sling with force energy, generating consistent bash damage throws. Inspired by wizard sling usage in Baldur's Gate 2.

Corrosive Aura: Acid suspended in a layer above your clothing, adding acid damage to your melee attacks & burning melee attackers.

Shardspray: Add metal component requirement, buff stats.

#### Describe alternatives you've considered

Playing Baldur's Gate 2.

#### Testing

Run around places that scrolls drop to see that the new ones appear.
Spawn every scroll & altered book to see that the correct spell is inside.
Look at both ethereal weapons for martial arts compatbility.
Cast every spell.

#### Additional context
The ITEM_DAMAGE_X are useless fields at the moment, due to not being covered by relic data yet.

### CODE REVIEW

File: data/mods/Magiclysm/items/spell_scrolls.json [hunk 1]
```diff
@@ -917,5 +917,61 @@
     "name": { "str": "Sappy Scroll" },
     "description": "This scroll is covered in sap and rose thorns.  You can still make out some of the writings, but it probably won't work the same.",
     "use_action": { "type": "learn_spell", "spells": [ "druid_tornskin" ] }
+  },
+  {
+    "type": "BOOK",
+    "copy-from": "spell_scroll",
+    "id": "spell_scroll_feralform",
+    "name": { "str": "Scroll of Feral Form" },
```
Comment (Qrox) [inline data/mods/Magiclysm/items/spell_scrolls.json:925]:
```suggestion
    "name": { "str": "Scroll of Feral Form", "str_pl": "Scrolls of Feral Form" },
```
Same for other scrolls.

File: data/mods/Magiclysm/Spells/biomancer.json [hunk 2]
```diff
@@ -272,5 +272,49 @@
...
+    "max_range": 2,
+    "min_dot": 2,
+    "max_dot": 2,
+    "min_duration": 300,
+    "max_duration": 300
+  },
+  {
+    "id": "biomancer_corrosive_aura",
+    "type": "SPELL",
+    "name": "Corrosive Aura",
```
Comment (KorGgenT) [inline data/mods/Magiclysm/Spells/biomancer.json:299]: image
spawns 0 auras

File: data/mods/Magiclysm/Spells/biomancer.json [hunk 3]
```diff
@@ -272,5 +272,49 @@
     "min_duration": 6000,
     "max_duration": 12000,
     "duration_increment": 300
+  },
+  {
+    "id": "acidic_zap",
+    "type": "SPELL",
+    "name": "Corrosive Aura",
```
Comment (KorGgenT) [inline data/mods/Magiclysm/Spells/biomancer.json:279]: i'd change this for debug reasons

File: data/mods/Magiclysm/items/ethereal_items.json [hunk 4]
```diff
@@ -475,5 +475,146 @@
...
     "flags": [ "AURA", "ONLY_ONE", "TRADER_AVOID", "NO_TAKEOFF", "NONCONDUCTIVE", "OVERSIZE" ],
     "relic_data": { "passive_effects": [ { "has": "WORN", "condition": "ALWAYS", "hit_me_effect": [ { "id": "thorns_zap" } ] } ] }
+  },
+  {
+    "id": "aura_feral",
+    "type": "ARMOR",
+    "name": "feral aura",
+    "description": "The manifestion of your rage.  It won't disappear until you calm down.",
+    "weight": "1g",
+    "volume": "1ml",
```
Comment (KorGgenT) [inline data/mods/Magiclysm/items/ethereal_items.json:485]:
```suggestion
    "weight": "1 g",
    "volume": "1 ml",
```

File: data/mods/Magiclysm/Spells/technomancer.json [hunk 5]
```diff
@@ -480,5 +480,57 @@
...
+    "max_damage": 70,
+    "damage_increment": 2.0,
+    "damage_type": "stab",
+    "min_range": 10,
+    "max_range": 20,
+    "range_increment": 0.5,
+    "min_pierce": 3,
+    "max_pierce": 10,
+    "pierce_increment": 0.35,
+    "learn_spells": { "technomancer_knifeshot_superior": 20 }
```
Comment (KorGgenT) [inline data/mods/Magiclysm/Spells/technomancer.json:511]: i don't seem to be gaining the new spell when i get to level 20.

Comment (LaVeyanFiend) [reply data/mods/Magiclysm/Spells/technomancer.json:511]: This also applies to the Acid Resistance aura not giving Greater Acid Resistance anymore, I'm not sure how it's breaking.

Comment (KorGgenT) [reply data/mods/Magiclysm/Spells/technomancer.json:511]: ok i will investigate separately and won't let it block this PR.

**Pull request merged**